Add SetAuthenticationType to TestAuthorizationContext (#650)
bUnit-dev/bUnit: docs/site/docs/test-doubles/faking-auth.md

[!code-csharp[UserRightsTest.cs](../../../samples/tests/xunit/UserRightsTest.cs?start=129&end=147&highlight=4-8)]

With this example done, all auth-related test scenarios should be covered. If you find that one is missing, please let us know in the [bUnit discussion forum](https://github.com/egil/bUnit/discussions).

### Authentication Types

> To specify one or more authenticaty type for the authenticated and authorized user, do the following:

Reviewer: ```suggestion
To specify one or more authentication types for the authenticated and authorized user, do the following:
```

Reviewer: Can you specify more than one authentication type?

Author: I'm thinking yes, like Cookie Auth, multiple OAuth providers, etc.

> ### Authentication Types
> 
> To specify one or more authentication types for the authenticated and authorized user, do the following:

Reviewer: ```suggestion
### Authentication types

To specify a authentication type for the authenticated and authorized user, do the following:
```

I guess its only possible to set a single authentication type at once, right? If so, I think this wording is more correct.

> ### Authentication Types
> 
> To specify one or more authentication types for the authenticated and authorized user, do the following:
> 
> [!code-csharp[UserRightsTest.cs](../../../samples/tests/xunit/UserRightsTest.cs?start=153&end=166&highlight=5)]
> 
> The highlighted line shows how the <xref:Bunit.TestDoubles.TestAuthorizationContext.SetAuthenticationType(System.String)> method is used to change the `Identity.AuthenticationType` of the user.

Reviewer: ```suggestion
### Authentication types

To specify a authentication type for the authenticated and authorized user, do the following:

[!code-csharp[UserRightsTest.cs](../../../samples/tests/xunit/UserRightsTest.cs?start=153&end=166&highlight=5)]

The highlighted line shows how the <xref:Bunit.TestDoubles.TestAuthorizationContext.SetAuthenticationType(System.String)> method is used to change the `Identity.AuthenticationType` of the user.
```
